sympy/solvers/ode.py

```diff
@@ -1044,6 +1037,47 @@ class in it.  Note that a hint may do this anyway if
             else:
                 raise ValueError("Enter boundary conditions of the form ics={f(point}: value, f(x).diff(x, order).subs(x, point): value}")
 
+
+    # Precondition to try remove unnecessary factor and power from equation
+    eqs = eq
+    if isinstance(eqs, Pow):
+        eqs_args = eqs.args # if f(x)**p=0 then f(x)=0 (p>0)
+        if eqs_args[1]>0:
+            eqs = eqs_args[0]
+
+    # I will try not to change the structure of equation may be user wants to solve the ode by some particular classifier
+    def _finding_algebraic_ode_factor(eqs, func, frac, hard_factor=False):
+
+        from sympy.core.exprtools import factor_terms
+        from sympy.polys.polytools import factor
+
+        if hard_factor:
+            eqs = factor(eqs)
+        eqs = factor_terms(eq, clear=False, fraction=frac)
+        eqs = fraction(eqs)[0] # p/q =0, So we need to solve only p=0
+        eqs_diff = [] # factor containing differential
+        eqs_algebraic = [] # factor containing only algebraic
+        if isinstance(eqs, Mul) and eqs.has(func):
+            fac = eqs.args
+            for i in fac:
+                if i.has(func) and i.has(Derivative):
+                    eqs_diff.append(i)
+                elif i.has(func):
+                    eqs_algebraic.append(i)
+        return eqs_diff, eqs_algebraic
+
+    eqs_diff, eqs_algebraic = _finding_algebraic_ode_factor(eqs, func, frac=True, hard_factor=True) # for factorable type classifier only
```

Comment:
You should limit the length of long lines of code. There can sometimes be a reason for allowing extra long lines but not just to fit a comment at the end (put it on the line above).

Reply:
okay